src/main/java/org/prebid/server/bidder/rubicon/RubiconBidder.java

```diff
@@ -401,19 +412,26 @@ private Imp makeImp(Imp imp,
         final App app = bidRequest.getApp();
         final Site site = bidRequest.getSite();
         final ExtRequest extRequest = bidRequest.getExt();
+        final boolean isVideo = isVideo(imp);
+
+        final PriceFloorResult priceFloorResult = isVideo ? resolvePriceFloors(bidRequest, imp) : null;
 
         final Imp.ImpBuilder builder = imp.toBuilder()
                 .metric(makeMetrics(imp))
-                .ext(mapper.mapper().valueToTree(makeImpExt(imp, extImpRubicon, site, app, extRequest)));
+                .ext(mapper.mapper().valueToTree(
+                        makeImpExt(imp, extImpRubicon, site, app, extRequest, priceFloorResult)));
+
+        final BigDecimal resolvedBidFloor = priceFloorResult != null
+                ? priceFloorResult.getFloorValue()
```

Comment:
Floor resolver returned floor in EUR, than without any currency conversion you sets USD currency on 430 line